packages/askui-nodejs/src/utils/proxy/proxy-utils.ts

```diff
@@ -0,0 +1,111 @@
+import fs from 'fs';
+import http from 'http';
+import https from 'https';
+import { join } from 'path';
+import dns, { LookupOneOptions } from 'dns';
+
+const proxy = require('proxy');
+
+export const PROXY_HOSTNAME = 'proxy.proxy-unit-test-askui.com';
+export const SERVER_HOSTNAME = 'server.proxy-unit-test-askui.com';
+
+/** *
+ * This file is oriented by the integration tests of hpagent: https://github.com/delvedor/hpagent/blob/main/test/utils.js
+ */
+export function addBasicAuthentication(httpProxy: http.Server): http.Server {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any, no-param-reassign
+  (httpProxy as any).authenticate = (req: any, fn: any) => {
+    const header = req.headers['proxy-authorization'] || ''; // get the auth header
+    const token = header.split(/\s+/).pop() || ''; // and the encoded auth token
+    const auth = Buffer.from(token, 'base64').toString(); // convert from base64
+    const parts = auth.split(/:/); // split on colon
+    const username = parts.shift(); // username is first
```

Comment:
can we pls. remove all the comments, I don't see that they bring any value as they say exactly what is done in code